tools/project_manager/show_image_sources.php

```diff
@@ -197,71 +158,20 @@
         echo "<td>{$row['public_comment']}</td>";
         echo "</tr>\n";
 
-        // Preserving the if ($logged_in) logic may seem pointless, but
-        // in the event we decide to remove the login requirement at the
-        // top of the file, we'll want the same behavior back.
-        // The former behavior was:
-        // if ($logged_in)
-        //     sort by display_name and include the internal_comment
-        // else
-        //     sort by the full_name and do not display internal_comment
-        // For now, we'll simply suppress the internal_comment.
-
         echo "<tr>\n";
         echo "<th class='label'>" . _("Notes") . ":</th>\n";
         echo "<td>";
-        if ($logged_in) {
-            echo html_safe($row['internal_comment']);
-        }
+        echo html_safe($row['internal_comment']);
         echo "</td>";
         echo "</tr>\n";
     }
 
     echo "</table>\n";
 } else {
-    $imso_code = $_GET['name'];
+    $image_sources = load_image_sources();
+    $imso = $image_sources[$imso_code] ?? [];
 
-    $imso = mysqli_fetch_assoc(mysqli_query(DPDatabase::get_connection(), sprintf("
-        SELECT
-            full_name,
-            display_name,
-            public_comment,
-            internal_comment,
-            info_page_visibility,
-            concat('<a href=\"',url,'\">',url,'</a>') as 'info_url'
-        FROM image_sources
-        WHERE code_name = '%s'
-    ", mysqli_real_escape_string(DPDatabase::get_connection(), $imso_code))));
-
-    $visibility = $imso['info_page_visibility'];
-
-
-    // info page visibility
-    //  0 = Image Source Managers and SAs
-    //  1 = also any PM
-    //  2 = also any logged-in user
-    //  3 = anyone
-
-    // layout below intended to make logic easier to follow:
-    // see it as a tree lying on it's side, the root node the "or"
-    // on the third line
-
-    $can_see = (
-                ($visibility == 3)
-            or
-                ($logged_in
-                    and
-                    (
-                        $visibility == 2
-                        or
-                        (user_is_PM() and $visibility == 1)
-                        or
-                        user_is_image_sources_manager()
-                    )
-                )
-            );
-
-    if ($can_see) {
+    if ($imso && can_user_see_image_source($imso)) {
```

Comment:
This one, too, maybe? The original code seems to be aimed at just determining whether or not the user is logged in, not excluding PMs from being able to select an enabled image source that excludes a PM from being able to access the images after the project is archived.

All the above being taken into account, I may be wrong in my interpretation, so we may need further research.

Reply:
None of the image source code seems to be related to archived projects. Can you elaborate more on how we think they're related.

Reply:
I was totally thinking of something else -- my apologies.

Based on investigation, Pending and Disabled projects are not displayed as select options for the PMs to choose from, whatever their visibility setting, but they are shown in the display list unless they've been set to ISM only.